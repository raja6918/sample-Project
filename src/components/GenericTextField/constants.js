export const TEXT_FIELD_REGEX = '^[a-zA-Z0-9].*';
export const TEXT_FIELD_ERROR = 'ERRORS.GENERIC_TEXT_FIELD.name';
