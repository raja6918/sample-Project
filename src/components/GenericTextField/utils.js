export const evaluateRegex = (regex, term) => {
  const reg = new RegExp(regex);
  const isValidRegex = reg.test(term);
  return isValidRegex;
};
