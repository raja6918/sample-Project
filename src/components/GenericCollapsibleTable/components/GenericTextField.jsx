import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';
import ResetButton from './ResetButton';
import { ERROR_ZINDEX } from './constants';
import './style.scss';

export default class GenericTextField extends Component {
  state = {
    value:
      this.props.value === null || this.props.value === undefined
        ? ''
        : this.props.value,
    editing: false,
  };

  componentWillReceiveProps(nextProps) {
    // To revert to previous value if the API call failed
    if (
      nextProps.value !== this.state.value &&
      !nextProps.error &&
      !this.state.editing
    ) {
      this.setState({
        value:
          nextProps.value === null || nextProps.value === undefined
            ? ''
            : nextProps.value,
      });
    }
  }

  handleOnBlur = () => {
    const value = this.props.converter(this.state.value);
    if (value !== this.props.value) {
      if (!value && value !== 0) {
        this.setState({ value: this.props.value, editing: false }, () => {
          this.props.handleRemoveParamSet(this.props.data);
        });
      } else {
        this.setState({ editing: false }, () => {
          this.props.onChange(value, this.props.data);
        });
      }
    } else {
      this.setState({ editing: false });
    }
  };

  handleOnChange = event => {
    if (event.target.value.length > this.props.maxInputLength) {
      event.target.value = event.target.value.slice(
        0,
        this.props.maxInputLength
      );
    }
    if (
      event.target.value === '' ||
      this.props.pattern.test(event.target.value)
    ) {
      const { value } = event.target;
      this.setState({ value });
    }
  };

  generateStyle = () => {
    let width;
    if (this.state.value === '' && this.props.placeholder) {
      width = this.props.placeholder.length;
    } else {
      const value =
        this.state.value === null || this.state.value === undefined
          ? ''
          : this.state.value;
      width =
        value.toString().length <= this.props.maxInputLength
          ? value.toString().length + 2
          : 9;
    }

    const zIndex = this.props.error ? ERROR_ZINDEX : 1;
    return { width: width + 'ch', zIndex, ...this.props.style };
  };

  removeOverlay = () => {
    this.setState({ editing: true }, () => {
      // To remove any outer overlay if any when we use this component
      this.props.removeOverlay();
    });
  };

  render() {
    const {
      value,
      data,
      error,
      handleDisable,
      type,
      rightPadding,
      enableReset,
      handleReset,
      handleTooltipDisable,
      getTooltipContent,
      placeholder,
    } = this.props;
    const disabled = handleDisable(value, data);
    const tooltipDisabled = handleTooltipDisable(value, data);
    const tooltipContent = getTooltipContent(value, data);

    return (
      <span className="rule-description-input">
        {' '}
        <TextField
          error={error}
          type={type}
          style={this.generateStyle()}
          disabled={disabled}
          onBlur={this.handleOnBlur}
          value={this.state.value}
          className={error ? 'input-error' : ''}
          onChange={this.handleOnChange}
          onFocus={event => event.target.select()}
          placeholder={placeholder}
          onClick={this.removeOverlay}
        />
        {rightPadding ? ` ${rightPadding} ` : ''}
        {enableReset && (
          <span className="reset-btn-container">
            (
            <ResetButton
              data={data}
              handleReset={handleReset}
              disabled={tooltipDisabled}
              tooltipContent={tooltipContent}
            />
            )
          </span>
        )}{' '}
      </span>
    );
  }
}

GenericTextField.propTypes = {
  data: PropTypes.shape().isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  style: PropTypes.shape(),
  handleDisable: PropTypes.func,
  handleReset: PropTypes.func,
  enableReset: PropTypes.bool,
  handleTooltipDisable: PropTypes.func,
  getTooltipContent: PropTypes.func,
  error: PropTypes.bool,
  type: PropTypes.string,
  maxInputLength: PropTypes.number,
  rightPadding: PropTypes.string,
  placeholder: PropTypes.string,
  pattern: PropTypes.instanceOf(RegExp),
  removeOverlay: PropTypes.func,
  converter: PropTypes.func,
  handleRemoveParamSet: PropTypes.func,
};

GenericTextField.defaultProps = {
  style: { color: '#ff650c' },
  error: false,
  handleDisable: () => false,
  enableReset: false,
  handleReset: () => null,
  handleTooltipDisable: () => false,
  getTooltipContent: () => false,
  type: 'text',
  maxInputLength: 7,
  rightPadding: '',
  placeholder: '',
  pattern: /./,
  removeOverlay: () => null,
  converter: value => value,
  handleRemoveParamSet: () => null,
};
