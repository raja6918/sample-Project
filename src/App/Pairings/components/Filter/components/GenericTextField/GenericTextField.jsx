import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';
import '../style.scss';
import { shortenText } from '../../../../../../utils/common';
import SierraTooltip from './../../../../../../_shared/components/SierraTooltip';

export default class GenericTextField extends Component {
  state = {
    value: this.props.value,
    editing: false,
    displayValue: '',
  };

  componentDidMount() {
    this.setState({
      displayValue: shortenText(this.state.value, this.props.maxDisplayLength),
    });
  }

  componentWillReceiveProps(nextProps) {
    // To revert to previous value if the API call failed
    if (
      nextProps.value !== this.state.value &&
      !nextProps.error &&
      !this.state.editing
    ) {
      this.setState({ value: nextProps.value });
    }
  }

  handleOnBlur = () => {
    const { value } = this.state;
    const { maxDisplayLength, onChange, converter } = this.props;
    this.setState(
      { editing: false, displayValue: shortenText(value, maxDisplayLength) },
      () => {
        const val = value ? converter(value) : '';
        onChange(val);
      }
    );
  };

  handleOnChange = event => {
    if (event.target.value.length > this.props.maxInputLength) {
      event.target.value = event.target.value.slice(
        0,
        this.props.maxInputLength
      );
    }
    if (
      event.target.value === '' ||
      this.props.pattern.test(event.target.value)
    ) {
      const { value } = event.target;
      this.setState({ value, displayValue: value });
    }
  };

  render() {
    const {
      value,
      error,
      handleDisable,
      type,
      style,
      placeholder,
      label,
    } = this.props;
    const { displayValue, value: stateValue } = this.state;
    const disabled = handleDisable(value);

    return (
      <span className="filter-panel-input">
        {' '}
        <SierraTooltip
          title={' '}
          disabled={stateValue === displayValue}
          position="bottom"
          html={<p style={{ padding: '10px' }}>{stateValue}</p>}
        >
          <TextField
            label={label}
            error={error}
            type={type}
            style={style}
            disabled={disabled}
            onBlur={this.handleOnBlur}
            value={displayValue}
            className={error ? 'input-error' : ''}
            onChange={this.handleOnChange}
            onFocus={event => {
              event.target.select();
              this.setState({ displayValue: stateValue });
            }}
            placeholder={placeholder}
            onClick={this.removeOverlay}
          />
        </SierraTooltip>
      </span>
    );
  }
}

GenericTextField.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  style: PropTypes.shape(),
  handleDisable: PropTypes.func,
  error: PropTypes.bool,
  type: PropTypes.string,
  label: PropTypes.string,
  maxInputLength: PropTypes.number,
  placeholder: PropTypes.string,
  pattern: PropTypes.instanceOf(RegExp),
  converter: PropTypes.func,
  maxDisplayLength: PropTypes.number,
};

GenericTextField.defaultProps = {
  style: {},
  error: false,
  handleDisable: () => false,
  type: 'text',
  label: '',
  maxInputLength: 7,
  placeholder: '',
  pattern: /./,
  converter: value => value,
  maxDisplayLength: 9,
};
