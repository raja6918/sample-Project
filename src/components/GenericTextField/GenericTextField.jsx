import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { InputAdornment, Icon, TextField } from '@material-ui/core';
import FormControlMUI from '@material-ui/core/FormControl';
import { evaluateRegex } from './utils';
import { TEXT_FIELD_REGEX, TEXT_FIELD_ERROR } from './constants';
import ErrorMessage from '../FormValidation/ErrorMessage';

const FormControl = styled(FormControlMUI)`
  width: 100%;
`;
/**
 * Props
 * id(string): id to be passed (required),
 * readOnly(bool): read only mode check (required),
 * value(string): default value of the textfield (required),
 * onBlur(func): callback on focus lost , returns id and value( required),
 * checkError(bool):checks to go for error handling, default false(optional)
 * regexString(string): regex pattern to check ,default ^[a-zA-Z0-9].* (optional),
 * errorMsg(string): error message for regex checking , default ERRORS.GENERIC_TEXT_FIELD.name(optional),
 * t(func): translate function for i18next checking (optional)
 * isActive(func):callback to check whether textfield is clicked
 */

class GenericTextField extends Component {
  state = {
    active: false,
    value: this.props.value,
    error: false,
    checkError: this.props.checkError ? this.props.checkError : false,
  };

  handleOnBlur = () => {
    const noChanges = this.state.value === this.props.value;
    if (noChanges) {
      this.setState({ active: false });
      this.props.onBlur();
      return;
    }

    let nextState = {
      active: false,
    };

    if (!this.state.value.length) {
      nextState = { ...nextState, value: this.props.value };
    } else {
      nextState = { ...nextState, value: this.state.value };
    }

    if (!this.state.error) {
      this.setState(
        nextState,
        this.props.onBlur(this.props.id, nextState.value)
      );
    } else {
      this.setState({
        value: this.props.value,
        error: false,
      });
    }
  };

  handleOnChange = event => {
    const { value } = event.target;
    if (this.state.checkError) {
      const regex = this.props.regexString
        ? this.props.regexString
        : TEXT_FIELD_REGEX;
      const isError = value === '' ? false : !evaluateRegex(regex, value);
      this.setState({ error: isError, value: value });
    } else {
      this.setState({ value });
    }
  };

  handleOnClick = () => {
    this.setState({ active: !this.state.active }, () => {
      this.props.isActive(this.state.active);
    });
  };

  componentWillReceiveProps(nextProps) {
    this.setState({ active: false, value: nextProps.value, error: false });
  }

  render() {
    const { active, value, error } = this.state;
    const { id, readOnly, t, errorMsg } = this.props;
    const errorMessage = errorMsg ? errorMsg : TEXT_FIELD_ERROR;
    let commonProps = {
      id,
      value,
      inputProps: {
        maxLength: 50,
      },
      onFocus: event => event.target.select(),
      onBlur: this.handleOnBlur,
      onChange: this.handleOnChange,
      disabled: readOnly,
      error: error,
    };

    if (!active) {
      commonProps = {
        ...commonProps,
        onClick: !readOnly ? this.handleOnClick : null,
        InputProps: {
          endAdornment: (
            <InputAdornment position="end">
              <Icon color={readOnly ? 'disabled' : 'primary'}>edit</Icon>
            </InputAdornment>
          ),
        },
      };
    }

    return (
      <FormControl error={error}>
        <TextField {...commonProps} />
        {this.state.checkError && (
          <ErrorMessage isVisible={error} message={t(errorMessage)} />
        )}
      </FormControl>
    );
  }
}

GenericTextField.propTypes = {
  id: PropTypes.string.isRequired,
  readOnly: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  checkError: PropTypes.bool,
  regexString: PropTypes.string,
  errorMsg: PropTypes.string,
  t: PropTypes.func,
  isActive: PropTypes.func,
};

export default GenericTextField;
