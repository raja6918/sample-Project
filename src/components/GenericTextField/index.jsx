import GenericTextField from './GenericTextField';

export default GenericTextField;
