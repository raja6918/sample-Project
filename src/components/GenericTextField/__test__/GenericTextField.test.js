import React from 'react';
import { mount } from 'enzyme';
import GenericTextField from '../GenericTextField';

const props = {
  onBlur: jest.fn(),
  value: 'solver title',
  readOnly: false,
  id: 'activeRequestName',
};

test.skip('Generic Test Component renders', () => {
  expect(mount(<GenericTextField {...props} />)).toMatchSnapshot();
});
