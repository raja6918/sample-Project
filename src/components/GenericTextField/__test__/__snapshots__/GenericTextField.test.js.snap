// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Test Component renders 1`] = `
.c0 {
  width: 100%;
}

<GenericTextField
  id="activeRequestName"
  onBlur={[MockFunction]}
  readOnly={false}
  value="solver title"
>
  <GenericTextField__FormControl
    error={false}
  >
    <WithStyles(ForwardRef(FormControl))
      className="c0"
      error={false}
    >
      <ForwardRef(FormControl)
        className="c0"
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth",
            "marginDense": "MuiFormControl-marginDense",
            "marginNormal": "MuiFormControl-marginNormal",
            "root": "MuiFormControl-root",
          }
        }
        error={false}
      >
        <div
          className="MuiFormControl-root c0"
        >
          <WithStyles(ForwardRef(TextField))
            InputProps={
              Object {
                "endAdornment": <UNDEFINED
                  position="end"
                >
                  <UNDEFINED
                    color="primary"
                  >
                    edit
                  </UNDEFINED>
                </UNDEFINED>,
              }
            }
            disabled={false}
            error={false}
            id="activeRequestName"
            inputProps={
              Object {
                "maxLength": 50,
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            value="solver title"
          >
            <ForwardRef(TextField)
              InputProps={
                Object {
                  "endAdornment": <UNDEFINED
                    position="end"
                  >
                    <UNDEFINED
                      color="primary"
                    >
                      edit
                    </UNDEFINED>
                  </UNDEFINED>,
                }
              }
              classes={
                Object {
                  "root": "MuiTextField-root",
                }
              }
              disabled={false}
              error={false}
              id="activeRequestName"
              inputProps={
                Object {
                  "maxLength": 50,
                }
              }
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              value="solver title"
            >
              <WithStyles(ForwardRef(FormControl))
                className="MuiTextField-root"
                color="primary"
                disabled={false}
                error={false}
                fullWidth={false}
                onClick={[Function]}
                required={false}
                variant="standard"
              >
                <ForwardRef(FormControl)
                  className="MuiTextField-root"
                  classes={
                    Object {
                      "fullWidth": "MuiFormControl-fullWidth",
                      "marginDense": "MuiFormControl-marginDense",
                      "marginNormal": "MuiFormControl-marginNormal",
                      "root": "MuiFormControl-root",
                    }
                  }
                  color="primary"
                  disabled={false}
                  error={false}
                  fullWidth={false}
                  onClick={[Function]}
                  required={false}
                  variant="standard"
                >
                  <div
                    className="MuiFormControl-root MuiTextField-root"
                    onClick={[Function]}
                  >
                    <WithStyles(ForwardRef(Input))
                      autoFocus={false}
                      endAdornment={
                        <UNDEFINED
                          position="end"
                        >
                          <UNDEFINED
                            color="primary"
                          >
                            edit
                          </UNDEFINED>
                        </UNDEFINED>
                      }
                      fullWidth={false}
                      id="activeRequestName"
                      inputProps={
                        Object {
                          "maxLength": 50,
                        }
                      }
                      multiline={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      value="solver title"
                    >
                      <ForwardRef(Input)
                        autoFocus={false}
                        classes={
                          Object {
                            "colorSecondary": "MuiInput-colorSecondary",
                            "disabled": "Mui-disabled",
                            "error": "Mui-error",
                            "focused": "Mui-focused",
                            "formControl": "MuiInput-formControl",
                            "fullWidth": "MuiInput-fullWidth",
                            "input": "MuiInput-input",
                            "inputMarginDense": "MuiInput-inputMarginDense",
                            "inputMultiline": "MuiInput-inputMultiline",
                            "inputTypeSearch": "MuiInput-inputTypeSearch",
                            "marginDense": "MuiInput-marginDense",
                            "multiline": "MuiInput-multiline",
                            "root": "MuiInput-root",
                            "underline": "MuiInput-underline",
                          }
                        }
                        endAdornment={
                          <UNDEFINED
                            position="end"
                          >
                            <UNDEFINED
                              color="primary"
                            >
                              edit
                            </UNDEFINED>
                          </UNDEFINED>
                        }
                        fullWidth={false}
                        id="activeRequestName"
                        inputProps={
                          Object {
                            "maxLength": 50,
                          }
                        }
                        multiline={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        value="solver title"
                      >
                        <WithStyles(ForwardRef(InputBase))
                          autoFocus={false}
                          classes={
                            Object {
                              "colorSecondary": "MuiInput-colorSecondary",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "focused": "Mui-focused",
                              "formControl": "MuiInput-formControl",
                              "fullWidth": "MuiInput-fullWidth",
                              "input": "MuiInput-input",
                              "inputMarginDense": "MuiInput-inputMarginDense",
                              "inputMultiline": "MuiInput-inputMultiline",
                              "inputTypeSearch": "MuiInput-inputTypeSearch",
                              "marginDense": "MuiInput-marginDense",
                              "multiline": "MuiInput-multiline",
                              "root": "MuiInput-root MuiInput-underline",
                              "underline": null,
                            }
                          }
                          endAdornment={
                            <UNDEFINED
                              position="end"
                            >
                              <UNDEFINED
                                color="primary"
                              >
                                edit
                              </UNDEFINED>
                            </UNDEFINED>
                          }
                          fullWidth={false}
                          id="activeRequestName"
                          inputComponent="input"
                          inputProps={
                            Object {
                              "maxLength": 50,
                            }
                          }
                          multiline={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="text"
                          value="solver title"
                        >
                          <ForwardRef(InputBase)
                            autoFocus={false}
                            classes={
                              Object {
                                "adornedEnd": "MuiInputBase-adornedEnd",
                                "adornedStart": "MuiInputBase-adornedStart",
                                "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                                "disabled": "Mui-disabled Mui-disabled",
                                "error": "Mui-error Mui-error",
                                "focused": "Mui-focused Mui-focused",
                                "formControl": "MuiInputBase-formControl MuiInput-formControl",
                                "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                                "input": "MuiInputBase-input MuiInput-input",
                                "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                                "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                                "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                                "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                                "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                                "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                                "multiline": "MuiInputBase-multiline MuiInput-multiline",
                                "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                              }
                            }
                            endAdornment={
                              <UNDEFINED
                                position="end"
                              >
                                <UNDEFINED
                                  color="primary"
                                >
                                  edit
                                </UNDEFINED>
                              </UNDEFINED>
                            }
                            fullWidth={false}
                            id="activeRequestName"
                            inputComponent="input"
                            inputProps={
                              Object {
                                "maxLength": 50,
                              }
                            }
                            multiline={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="text"
                            value="solver title"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                              onClick={[Function]}
                            >
                              <input
                                aria-invalid={false}
                                autoFocus={false}
                                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                                disabled={false}
                                id="activeRequestName"
                                maxLength={50}
                                onAnimationStart={[Function]}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                type="text"
                                value="solver title"
                              />
                              <WithStyles(ForwardRef(InputAdornment))
                                position="end"
                              >
                                <ForwardRef(InputAdornment)
                                  classes={
                                    Object {
                                      "disablePointerEvents": "MuiInputAdornment-disablePointerEvents",
                                      "filled": "MuiInputAdornment-filled",
                                      "hiddenLabel": "MuiInputAdornment-hiddenLabel",
                                      "marginDense": "MuiInputAdornment-marginDense",
                                      "positionEnd": "MuiInputAdornment-positionEnd",
                                      "positionStart": "MuiInputAdornment-positionStart",
                                      "root": "MuiInputAdornment-root",
                                    }
                                  }
                                  position="end"
                                >
                                  <div
                                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                                  >
                                    <WithStyles(ForwardRef(Icon))
                                      color="primary"
                                    >
                                      <ForwardRef(Icon)
                                        classes={
                                          Object {
                                            "colorAction": "MuiIcon-colorAction",
                                            "colorDisabled": "MuiIcon-colorDisabled",
                                            "colorError": "MuiIcon-colorError",
                                            "colorPrimary": "MuiIcon-colorPrimary",
                                            "colorSecondary": "MuiIcon-colorSecondary",
                                            "fontSizeInherit": "MuiIcon-fontSizeInherit",
                                            "fontSizeLarge": "MuiIcon-fontSizeLarge",
                                            "fontSizeSmall": "MuiIcon-fontSizeSmall",
                                            "root": "MuiIcon-root",
                                          }
                                        }
                                        color="primary"
                                      >
                                        <span
                                          aria-hidden={true}
                                          className="material-icons MuiIcon-root MuiIcon-colorPrimary"
                                        >
                                          edit
                                        </span>
                                      </ForwardRef(Icon)>
                                    </WithStyles(ForwardRef(Icon))>
                                  </div>
                                </ForwardRef(InputAdornment)>
                              </WithStyles(ForwardRef(InputAdornment))>
                            </div>
                          </ForwardRef(InputBase)>
                        </WithStyles(ForwardRef(InputBase))>
                      </ForwardRef(Input)>
                    </WithStyles(ForwardRef(Input))>
                  </div>
                </ForwardRef(FormControl)>
              </WithStyles(ForwardRef(FormControl))>
            </ForwardRef(TextField)>
          </WithStyles(ForwardRef(TextField))>
        </div>
      </ForwardRef(FormControl)>
    </WithStyles(ForwardRef(FormControl))>
  </GenericTextField__FormControl>
</GenericTextField>
`;
